src/mudlet.cpp

```diff
@@ -468,14 +469,6 @@ mudlet::mudlet()
     // value will be accepted:
     setToolBarIconSize(file_use_smallscreen.exists() ? 2 : 3);
 
-    // Recover from a save in a state when both are hidden so that the toolbar
-    // is shown
-    // TODO: Provide a main console context menu item to restore at least one
-    // of them - if both are hidden.
-    if (!(mShowMenuBar || mShowToolbar)) {
```

Comment:
Is code that does this still present somewhere?

Reply:
It is now cured _differently_ - note the comment above. 😁

Reply:
That is an incomplete fix - you can still hide the buttons by right-clicking on them and unticking the box. This 'emergency restore' needs to come back.

Reply:
Oh, yeah - have got a fix for that  - by coupling up the main toolkbar's `QToolbar::visibilityChanged(bool)` signal to a new slot that prevents it being hidden if it is not set to be - which it won't be if there is no way back...

Reply:
Oh, yeah - have got a fix for that  - by coupling up the main toolkbar's `QToolbar::visibilityChanged(bool)` signal to a new slot that prevents it being hidden if it is not set to be - which it won't be if there is no way back...